packages/api/lambdas/create-reconciliation-report.js

```diff
@@ -32,21 +36,94 @@ const {
   ReconciliationReportPgModel,
   translatePostgresReconReportToApiReconReport,
 } = require('@cumulus/db');
-const { createInternalReconciliationReport } = require('./internal-reconciliation-report');
 const { createGranuleInventoryReport } = require('./reports/granule-inventory-report');
 const { createOrcaBackupReconciliationReport } = require('./reports/orca-backup-reconciliation-report');
 const { errorify, filenamify } = require('../lib/utils');
 const {
   cmrGranuleSearchParams,
-  convertToESCollectionSearchParams,
-  convertToESGranuleSearchParams,
+  convertToDBGranuleSearchParams,
+  convertToDBCollectionSearchObject,
+  dateStringToDateOrNull,
   initialReportHeader,
 } = require('../lib/reconciliationReport');
 
 const log = new Logger({ sender: '@api/lambdas/create-reconciliation-report' });
 
 const isDataBucket = (bucketConfig) => ['private', 'public', 'protected'].includes(bucketConfig.type);
 
+// Typescript annotations
```

Comment:
Reviewers:  we could, in theory use a separate types file.    Current convention is annotations in-line for types that aren't shared, this PR introduces a common types.js file that's up for discussion as well.     Feedback on organization welcome.

Reply:
Ok, fair enough!  I'll get a commit here shortly.

Reply:
Take a look at: b7bd336235a

Reply:
Nice. Thanks!